relay-sampling/src/lib.rs

```diff
@@ -878,11 +878,32 @@ impl Display for MatchedRuleIds {
     }
 }
 
+/// Checks whether unsupported rules result in a direct keep of the event or depending on the
+/// type of Relay an ignore of unsupported rules.
+fn check_unsupported_rules(
+    processing_enabled: bool,
+    sampling_config: &SamplingConfig,
+    root_sampling_config: Option<&SamplingConfig>,
+) -> Option<()> {
```

Comment:
nit: I know this function was only moved, but maybe for a follow-up PR: I find a return type of `Option<()>` hard to read, I would prefer `Result<(), ()>`, and then map it to option with `.ok()` on the caller side.

Reply:
I can do that! Thanks for the suggestion